guides/keras_cv/generate_with_stable_diffusion.py

```diff
@@ -0,0 +1,294 @@
+)
+
+"""
+Or the Garden of Eden filled with cats?
+"""
+
+visualize_prompt(
+    "Absolutely stunning artwork of the garden of eden, " "but filled with cats!"
+)
+
+"""
+Or a cute magical flying dog with a lengthy list of descriptors:
+"""
+
+
+visualize_prompt(
+    "a cute magical flying dog, fantasy art drawn by disney concept artists, "
+    "golden colour, high quality, highly detailed, elegant, sharp focus, "
+    "concept art, character concepts, digital painting, mystery, adventure"
+)
+
+"""
+The possibilities are literally endless (or as you will learn later: at least extend to the boundaries of StableDiffusion's latent manifold).
+
+Pretty incredible!  The idea should be self evident at this point.
+Now lets take a step back and look at how this algorithm actually works.
+
+## The StableDiffusion Algorithm
+
+TODO(lukewood): write this
+"""
+# Need to write up the actual algorithm and provide an overview
+
+"""
+## Perks of KerasCV
+
+So, with numerous implementations of StableDiffusion publicly available why shoud you use `keras_cv.models.StableDiffusion()`?
+
+Aside from the easy-to-use API, KerasCV's StableDiffusion model comes with some nice bells and trinkets.  These extra features include but are not limited to:
+
+- out of the box support for XLA compilation
+- support for `mixed_precision`
+- ...
+
+When these are combined, the KerasCV StableDiffusion model runs orders of magnitude faster than naive implementations.  This section shows how to enable all of these features, and the resulting performance gain yielded from using them.
+
+**Note: The difference between the performance benefits from each optimization vary drastically between hardware**
+"""
+
+"""
+### XLA Compilation
+
+TensorFlow comes with the [XLA: Accelerated Linear Algebra](https://www.tensorflow.org/xla) compiler built in.
+`keras_cv.models.StableDiffusion` supports a `jit_compile` argument out of the box.
+Setting this argument to `True` enables XLA compilation; resulting in a significant speed-up.
+
+Lets use this below.
+"""
+
+xla_stable_diffusion = keras_cv.models.StableDiffusion(jit_compile=True)
+
+# warm up the model by running inference once before timing it
+visualize_prompt(
+    "An oldschool macintosh computer showing an avocado on its screen",
+    sd_model=xla_stable_diffusion,
+)
+
+"""
+Now lets compare the runtimes of our models
+"""
+
+with ez_timer() as timer:
+    visualize_prompt(
+        "A cute water-colored otter in a rainbow whirlpool holding shells",
+        sd_model=stable_diffusion,
+    )
+print(f"Without XLA took {timer.result} seconds")
+
+"""
+and now with XLA
+"""
+
+with ez_timer() as timer:
+    visualize_prompt(
+        "A cute water-colored otter in a rainbow whirlpool holding shells",
+        sd_model=xla_stable_diffusion,
+    )
+print(f"With XLA took {timer.result} seconds")
+
+"""
+On my hardware I see about a 2x speedup.
+
+### Mixed Precision
```

Comment:
I'd start with mixed precision intro right after generating the first batch of images. Then XLA. Then a table showing throughput in different configs and comparing it to PT.